src/impy/models/sibyll.py

```diff
@@ -142,61 +114,54 @@ def _set_stable(self, pdgid, stable):
             idb[sid - 1] = abs(idb[sid - 1])
 
     def _generate(self):
-        kin = self.kinematics
-        self._lib.sibyll(self._production_id, kin.p2.A, kin.ecm)
+        k = self._kin
+        self._lib.sibyll(self._production_id, k.p2.A, k.ecm)
         self._lib.decsib()
         self._lib.sibhep()
         return True
 
-    @property
-    def random_state(self):
-        return RMMARDSib()._record_state(self)
-
-    @random_state.setter
-    def random_state(self, rng_state):
-        rng_state._restore_state(self)
-
 
-class Sibyll21(SIBYLLRun):
+class Sibyll21(SIBYLLRun, MCRun):
     _version = "2.1"
     _library_name = "_sib21"
 
 
-class Sibyll23(SIBYLLRun):
+# For some reason, Sibyll23 requires MCRunRemote, but the others don't
```

Comment:
Does this mean that only Sibyll23 crashes on multiple inits? In SIBYLL, the initialization should not be called multiple times even if it works.

Reply:
Yes, that's exactly what it means, you can try it out. Replace MCRunRemote with MCRun further below as base class for Sibyll23 and run the tests.